stock_picking_return_restricted_qty/wizard/stock_picking_return.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2020 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class ReturnPicking(models.TransientModel):
+    _inherit = "stock.return.picking"
+
+    @api.model
+    def _prepare_stock_return_picking_line_vals_from_move(self, stock_move):
+        val = super()._prepare_stock_return_picking_line_vals_from_move(stock_move)
+        return_lines = self.env["stock.return.picking.line"]
+        val["quantity"] = return_lines.get_returned_restricted_quantity(stock_move)
+        return val
+
+    def _create_returns(self):
+        for return_line in self.product_return_moves:
+            quantity = return_line.get_returned_restricted_quantity(return_line.move_id)
+            if return_line.quantity > quantity:
+                raise UserError(
+                    _("Return more quantities than delivered is not allowed.")
+                )
+        return super()._create_returns()
+
+
+class ReturnPickingLine(models.TransientModel):
+    _inherit = "stock.return.picking.line"
+
+    @api.onchange("quantity")
+    def _onchange_quantity(self):
+        qty = self.get_returned_restricted_quantity(self.move_id)
+        if self.quantity > qty:
+            raise UserError(_("Return more quantities than delivered is not allowed."))
+
+    def get_returned_restricted_quantity(self, stock_move):
```

Comment:
@api.model ?

Reply:
Better to not restrict it and `self` can be used any time in the future.